### PR TITLE
Split cmd/otelcontribcol into a separate module, extract testbed in a separate module

### PR DESCRIPTION
**Pull request by bogdandrutu**
Updates https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/11867

### CODE REVIEW

File: processor/cumulativetodeltaprocessor/go.mod [hunk 1]
```diff
@@ -47,3 +47,5 @@ replace github.com/open-telemetry/opentelemetry-collector-contrib/internal/corei
 replace github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter => ../../internal/filter
 
 retract v0.65.0
+
+replace github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl => ../../pkg/ottl
```
Comment (bogdandrutu) [inline processor/cumulativetodeltaprocessor/go.mod:51]: Changes added automatically by `make crosslink`

**Comment by jpkrohling** [comment]
Why do we want to split the cmd/otelcontribcol into its own module? Isn't its sole purpose to help local development only?

**Comment by jpkrohling** [comment]
> To be able to make progress on https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/11867

Sorry, I'm still failing to see how this PR will help with that issue. Would you mind to expand on that?

**Comment by bogdandrutu** [comment]
> Sorry, I'm still failing to see how this PR will help with that issue. Would you mind expanding on that?

Sorry I have a hard time explaining the "builder" functionality...

1. The builder is capable of producing only one "binary" and cannot produce as we have right now 2 binaries based on the build flag "testbed". Does this make sense why we need 2 different cmd directories?
2. The builder produces a `go.mod` file, which means that it produces a module (not just a package inside a different go module), hence the need to split the cmds in separate go modules.

**Review by mx-psi — APPROVED** [review]

**Comment by jpkrohling** [comment]
> The builder produces a go.mod file, which means that it produces a module (not just a package inside a different go module), hence the need to split the cmds in separate go modules.

As clarified via Slack, the misunderstanding was that I wasn't expecting the generated code to be checked in, while @bogdandrutu has the intention of keeping the generated files, for auditing purposes. With that in mind, it does make sense then to have that as its own go module.

**Pull request merged**